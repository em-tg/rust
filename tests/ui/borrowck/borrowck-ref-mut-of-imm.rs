//@ check-pass
fn destructure(x: Option<isize>) -> isize {
    match x {
      None => 0,
      Some(ref mut v) => *v //~ WARNING cannot borrow
    }
}

fn main() {
    assert_eq!(destructure(Some(22)), 22);
}
