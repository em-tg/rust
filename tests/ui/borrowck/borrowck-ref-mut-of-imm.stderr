warning[E0596]: cannot borrow `x.0` as mutable, as `x` is not declared as mutable
  --> $DIR/borrowck-ref-mut-of-imm.rs:5:12
   |
LL |       Some(ref mut v) => *v
   |            ^^^^^^^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL | fn destructure(mut x: Option<isize>) -> isize {
   |                +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
